### PR TITLE
Issue #264: Fix for allocator property, notes about properties and outstanding_work.

### PR DESCRIPTION
**Pull request by AerialMantis**
**Summary**

* Add minor fixes to synopsis.
* Add fix for allocator property declaration.
* Add general note about the implementation requirements of properties.
* Add note about implications of the outstanding_work property.

### CODE REVIEW

**Comment by AerialMantis** [comment]
I've updated the pull request with the changes we discussed on Friday. I made one further change, rather than naming the allocator property struct `get_allocator_t` I named it `allocator_t` and I renamed the struct containing the allocator (originally `allocator_t`) to `allocator_wrapper_t` instead. My thinking is that the allocator property struct is used in the interface for both setting and getting so that should be named `allocator_t` while the struct containing the allocator is only used as a temporary container for the allocator. I also elaborated on the note about property implementation (now moved to the allocator requirements) as to what types the customization points should take.

@jaredhoberock this is ready to merge if everyone is happy with the changes.

**Comment by jaredhoberock** [comment]
Thanks Gordon, I like the name `allocator_t` as the name of the `allocator` object.

File: wording.md [hunk 1]
```diff
@@ -383,6 +383,8 @@ The `not_continuation` and `continuation` properties are mutually exclusive.
 
 The `not_outstanding_work` and `outstanding_work` properties are mutually exclusive.
 
+[*Note:* The `outstanding_work` and `not_outstanding_work` properties are use to communicate to the associated execution context intended future work submission on the executor. The intended effect of the properties is the behavior of execution context's facilities for awaiting outstanding work; specifically whether it considers's the existance of the executor object with the `outstanding_work` property enabled outstanding work when deciding what to wait on. However this will be largely defined by the execution context implementation. It is intended that the execution context will define it's wait facilities and on-destruction behaviour and provide an interface for querying this. An initial work towards this is included in P0737r0. *--end note*]
```
Comment (jaredhoberock) [inline wording.md]: Typo: considers's -> considers

Comment (jaredhoberock) [reply wording.md]: it's -> its

**Comment by jaredhoberock** [comment]
The changes look OK to me, modulo a couple of typos I noted.

**Comment by AerialMantis** [comment]
Great thanks, i've pushed a fix for the typos.

**Comment by jaredhoberock** [comment]
Thanks. Merged.

**Pull request merged**